core/src/main/java/com/graphhopper/routing/ch/NodeBasedNodeContractor.java

```diff
@@ -131,29 +137,52 @@ public IntContainer contractNode(int node) {
      * with them.
      */
     private void insertShortcuts(int node) {
-        shortcutHandler.startContractingNode();
+        shortcuts.clear();
         {
```

Comment:
Can we remove those code blocks?

Reply:
We could, but I think they are useful to limit the scope of the `iter` variable. But anyway I put both blocks into separate methods.